mergify_engine/tests/functional/actions/test_queue.py

```diff
@@ -223,6 +224,149 @@ async def assert_queued():
+        await self.run_engine()
+
+        pulls = await self.get_pulls()
+        assert len(pulls) == 7
+
+        tmp_pulls = sorted(
+            [
+                tmp
+                for tmp in pulls
+                if tmp["number"]
+                not in (
+                    p1["number"],
+                    p2["number"],
+                    p3["number"],
+                    p4["number"],
+                    p5["number"],
+                    p["number"],
+                )
+            ],
+            key=operator.itemgetter("number"),
+        )
+
+        ctxt = context.Context(self.repository_ctxt, p)
+        q = await merge_train.Train.from_context(ctxt)
+        await self._assert_cars_contents(
+            q,
+            p["merge_commit_sha"],
+            [
+                TrainCarMatcher(
+                    [p1["number"], p2["number"]],
+                    [],
+                    p["merge_commit_sha"],
+                    "created",
+                    tmp_pulls[0]["number"],
+                ),
+                TrainCarMatcher(
+                    [p3["number"], p4["number"]],
+                    [p1["number"], p2["number"]],
+                    p["merge_commit_sha"],
+                    "created",
+                    tmp_pulls[1]["number"],
+                ),
+            ],
+            [p5["number"]],
+        )
+
+        await self.create_status(tmp_pulls[0])
+        await self.run_engine()
+
+        pulls = await self.get_pulls()
+        assert len(pulls) == 5
+
+        tmp_pulls = sorted(
+            [
+                tmp
+                for tmp in pulls
+                if tmp["number"]
+                not in (
+                    p1["number"],
+                    p2["number"],
+                    p3["number"],
+                    p4["number"],
+                    p5["number"],
+                    p["number"],
+                )
+            ],
+            key=operator.itemgetter("number"),
+        )
+
+        p2 = await self.get_pull(p2["number"])
+        await self._assert_cars_contents(
+            q,
+            p2["merge_commit_sha"],
+            [
+                TrainCarMatcher(
+                    [p3["number"], p4["number"]],
+                    [p1["number"], p2["number"]],
+                    p["merge_commit_sha"],
+                    "created",
+                    tmp_pulls[0]["number"],
+                ),
+                TrainCarMatcher(
+                    [p5["number"]],
+                    [p3["number"], p4["number"]],
+                    p2["merge_commit_sha"],
+                    "created",
+                    tmp_pulls[1]["number"],
+                ),
+            ],
+        )
+
+        await self.create_status(tmp_pulls[0])
+        await self.create_status(tmp_pulls[1])
+        await self.run_engine()
+
+        pulls = await self.get_pulls()
+        assert len(pulls) == 0
+
+        await self._assert_cars_contents(q, None, [])
+
```

Comment:
Could we add a functional scenario for failure with a split?
That'd be really great to cover it here too.